core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -263,9 +263,32 @@ private[sql] final case class ColumnTableScan(
     }
   }
 
+  def splitMoveNextMethods(ctx: CodegenContext, blocks: ArrayBuffer[String],
+      arg: String): String = {
+    if (blocks.length == 1) {
+      // inline execution if only one block
+      blocks.head
+    } else {
+      val apply = ctx.freshName("moveNext")
+      val functions = blocks.zipWithIndex.map { case (body, i) =>
+        val name = s"${apply}_$i"
+        val code =
+          s"""
+             |private void $name(int $arg) {
```

Comment:
These are very customized functions. Not sure what value it will add to CodeGenContext unless it can be very generic in nature.